cuqi/distribution/_lmrf.py

```diff
@@ -39,35 +40,52 @@ class LMRF(Distribution):
     .. code-block:: python
 
         import cuqi
-        import numpy as np
-        prior = cuqi.distribution.LMRF(location=np.zeros(128), scale=0.1)
+        prior = cuqi.distribution.LMRF(location=0, scale=0.1, dim=128)
  
     """
-    def __init__(self, location, scale, bc_type="zero", physical_dim=1, **kwargs):
+    def __init__(self, location, scale, bc_type="zero", physical_dim=None, **kwargs):
         # Init from abstract distribution class
-        super().__init__(**kwargs) 
+        super().__init__(**kwargs)
 
         self.location = location
         self.scale = scale
         self._bc_type = bc_type
+
+        # Ensure geometry has shape
+        if not self.geometry.fun_shape or self.geometry.par_dim == 1:
```

Comment:
Just wondering if this should be `and` condition instead of `or` (same above in line 60 in _cmrf.py). The if statement does not seem to match `must be initialized with geometry or dim greater than 1.`. Could be something I am missing.

Another related point, FEniCS geometries has fun_shape None, so this error will be raised always.

Reply:
Added more details in error statement